claimer/src/claim-withdrawals.ts

```diff
@@ -1,8 +1,66 @@
 import { RelayVaultService } from '@relay-protocol/client'
+import { gql } from 'graphql-request'
+import networks from '@relay-protocol/networks'
+import { ethers } from 'ethers'
+import { claimFunds } from './relay'
 
-// Take all transactions that are initiated and attempts to prove them!
-export const claimTransactions = async (_params: {
+const GET_POOLS_AND_PROXY_BRIDGES = gql`
+  query {
+    relayPools {
+      items {
+        chainId
+        contractAddress
+        asset
+        origins {
+          items {
+            originChainId
+            originBridge
+            proxyBridge
+          }
+        }
+      }
+    }
+  }
+`
+
+// Checks a balance of ETH or ERC20
+const getBalance = async (chainId: number, wallet: string, asset: string) => {
+  const network = networks[chainId]
+  const provider = new ethers.JsonRpcProvider(network.rpc[0])
+  if (asset.toLowerCase() == network.assets.weth.toLowerCase()) {
+    return provider.getBalance(wallet)
```

Comment:
here we check for weth address but use the provider native balance, so I don't think that will return the correct balance

Reply:
> here we check for weth address but use the provider native balance, so I don't think that will return the correct balance

Well yes in fact. When the pool's asset is WETH, the `bridgeProxy` receives `ETH`... so we need to check its balance of WETH! The `claim` function handles the logic of claiming and receving eTH and wraps it!